manager/controllers/baseline_update.go

```diff
@@ -89,22 +93,22 @@ func BaselineUpdateHandler(c *gin.Context) {
 	newAssociations, obsoleteAssociations := sortInventoryIDs(req.InventoryIDs)
 	err = buildUpdateBaselineQuery(baselineID, req, newAssociations, obsoleteAssociations, account)
 	if err != nil {
-		LogAndRespError(c, err, "Database error")
-		return
+		switch e := err.(type) {
+		case *pgconn.PgError:
+			if e.Code == errDuplicatedKeyVal {
+				LogAndRespBadRequest(c, err, "baseline name already exists")
+				return
+			}
+		default:
+			LogAndRespError(c, err, "Database error")
+			return
+		}
```

Comment:
This is looks like slightly modified copy-paste of `baseline_create.go:76`.
Duplicated code mostly means it should be a function (somewhere in database/utils in this case) called on two places ;-).

Reply:
Could you propose what you have on your mind?

Reply:
I just noticed however potential bug there when error is of type pgconn.PgError and code is different. Will fix that.